src-tauri/src/tray.rs

```diff
@@ -0,0 +1,94 @@
+use ksni::blocking::TrayMethods;
+use tauri::{App, AppHandle, Manager, Runtime};
+use tauri::tray::{MouseButton, MouseButtonState, TrayIconBuilder, TrayIconEvent};
+
+#[derive(Debug)]
+struct TchapTray<R: Runtime> {
+    app_handle: AppHandle<R>,
+}
+
+impl<R: Runtime> TchapTray<R> {
+    fn new<M: Manager<R>>(manager: &M) -> Self {
+        Self { app_handle: manager.app_handle().clone() }
+    }
+}
+
+impl<R: Runtime> ksni::Tray for TchapTray<R> {
+    fn id(&self) -> String {
+        String::from("Tchap")
+    }
+
+    fn activate(&mut self, _x: i32, _y: i32) {
+        toggle_activation(&self.app_handle)
+    }
+
+    fn icon_pixmap(&self) -> Vec<ksni::Icon> {
+        let icon = self.app_handle.default_window_icon().unwrap();
+        let rgba = icon.rgba().to_vec();
+
+        // convert from rgba to argb
+        let mut bytes = rgba;
+        for i in 0..(bytes.len() / 4) {
+            let j = i * 4;
+            let a = bytes[j + 3];
+            bytes[j + 3] = bytes[j + 2];
+            bytes[j + 2] = bytes[j + 1];
+            bytes[j + 1] = bytes[j];
+            bytes[j] = a;
+        }
```

Comment:
This is from https://github.com/tauri-apps/tray-icon/pull/201/files#diff-9a3c7d84204fa3afdd13fb7d471f5208798a04165e981351c5b7080145538547R15-R38.